api_app/script_analyzers/file_analyzers/mwdb_scan.py

```diff
@@ -0,0 +1,46 @@
+import hashlib
+import time
+import mwdblib
+
+from api_app.exceptions import AnalyzerRunException
+from api_app.helpers import get_binary
+from api_app.script_analyzers.classes import FileAnalyzer
+
+
+class MWDB_Scan(FileAnalyzer):
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get("api_key_name", None)
+        self.upload_file = additional_config_params.get("upload_file", False)
+        self.max_retries = additional_config_params.get("max_retries", 20)
+
+    def run(self):
+        if not self.api_key_name:
+            raise AnalyzerRunException(
+                f"No API key retrieved with name: {self.api_key_name}"
+            )
+
+        mwdb = mwdblib.MWDB(api_key=self.api_key_name)
+        binary = get_binary(self.job_id)
+        query = str(hashlib.sha256(binary).hexdigest())
+
+        if self.upload_file:
+            file_object = mwdb.upload_file(query, binary)
+            file_object.flush()
+            while self.max_retries:
+                self.max_retries -= 1
+                time.sleep(10)
```

Comment:
see other analyzers on how they manage the polling.

`poll_distance` parameter can be found in a lot of other analyzers. The best way to write these new analyzers in a good way is to take examples from the others. See this: https://github.com/intelowlproject/IntelOwl/blob/7463881edc8448fd6f9c78a8ca192412c1a49578/api_app/script_analyzers/file_analyzers/triage_scan.py#L47-L54

Also, I would reduced the value from 10 to 5